app/src/androidTest/java/com/sdp/movemeet/map/MainMapFragmentTest.java

```diff
@@ -82,19 +117,115 @@ public void onComplete(@NonNull Task<AuthResult> task) {
+        assertEquals(0, defaultLocation.getLongitude(), 0);
+        assertEquals(0, defaultLocation.getLatitude(), 0);
+    }
+
+
 
     @Test
-    public void mainMapFragment_isDisplayed() throws InterruptedException {
-        onView(withId(R.id.fragment_map)).check(matches((isDisplayed())));
+    public void activitiesUpdatesOnAdd() {
+        MainMapFragment mapFragment = fragmentTestRule.getFragment();
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        ActivitiesUpdater updater = ActivitiesUpdater.getInstance();
+        BackendActivityManager bam = new BackendActivityManager(db, "activities");
+
+        updater.fetchListActivities();
+        updater.updateListActivities(mapFragment);
+
+        Activity act = new Activity("activity",
+                "me",
+                "title",
+                10,
+                new ArrayList<String>(),
+                0,
+                0,
+                "desc",
+                new Date(),
+                10,
+                Sport.Running,
+                "address",
+                new Date());
+
+        bam.uploadActivity(act,
+                new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        updater.updateListActivities(mapFragment);
+                        try {
+                            Thread.sleep(2000);
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                        }
+                        List<Activity> activities = updater.getActivities();
+                        Activity act_in_collection = activities.get(activities.size()-1);
+                        assertEquals(act.getActivityId(), act_in_collection.getActivityId());
+                        assertEquals(act.getAddress(), act_in_collection.getAddress());
+                        assertEquals(act.getDate(), act_in_collection.getDate());
+                        assertEquals(act.getDescription(), act_in_collection.getDescription());
+                        assertEquals(act.getLatitude(), act_in_collection.getLatitude());
+                        assertEquals(act.getLongitude(), act_in_collection.getLongitude());
+                        assertEquals(act.getNumberParticipant(), act_in_collection.getNumberParticipant());
+                        assertEquals(act.getParticipantId(), act_in_collection.getParticipantId());
+                        assertEquals(act.getTitle(), act_in_collection.getTitle());
+                        assertEquals(act.getOrganizerId(), act_in_collection.getOrganizerId());
+                        assertEquals(act.getSport(), act_in_collection.getSport());
+                    }
+                },
+                new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                    }
+                });
+
+    }
+
+
+    /*@Test
```

Comment:
Why this test is comment ?

Reply:
For some reason, it made the process crash. I couldn't find the reason, so I just commentend it for now. I intend to look at what the problem could be more in detail in the next couple of days 👍